packages/api/models/granules.js

```diff
@@ -673,9 +636,31 @@ class Granule extends Manager {
    * @returns {Promise}
    */
   async storeGranulesFromCumulusMessage(cumulusMessage) {
-    const granuleRecords = await this.constructor
-      ._getGranuleRecordsFromCumulusMessage(cumulusMessage);
-    return Promise.all(granuleRecords.map(this._validateAndStoreGranuleRecord, this));
+    const granules = getMessageGranules(cumulusMessage);
+    if (!granules) {
+      log.info(`No granules to process in the payload: ${JSON.stringify(cumulusMessage.payload)}`);
+      return [];
+    }
+
+    const executionArn = getMessageExecutionArn(cumulusMessage);
+    const executionUrl = getExecutionUrlFromArn(executionArn);
+
+    let executionDescription;
+    try {
+      executionDescription = await StepFunctions.describeExecution({ executionArn });
+    } catch (error) {
+      log.error(`Could not describe execution ${executionArn}`, error);
+    }
+
+    return Promise.all(granules.map(
```

Comment:
Do we want this to abort on some of the granules if one granule fails validation in storeGranule?   

More importantly given this is async, if we *do* want that to occur, are we ok with a random set of them succeeding? 

Consider changing this to an allSettled with evaluation to throw if a failure occurred.